src/traces/scatter/select.js

```diff
@@ -40,6 +41,11 @@ module.exports = function selectPoints(searchInfo, polygon) {
             di = cd[i];
             x = xa.c2p(di.x);
             y = ya.c2p(di.y);
+
+            if(x === BADNUM || y === BADNUM) {
+                continue;
+            }
```

Comment:
I take it from this fix that `polygon.contains` fails for undefined inputs? I would think for future extensibility it would be better to fix it in there, so if either `x` or `y` is `BADNUM` it's not contained in any polygon.